modules/core/src/main/scala/org/scalasteward/core/application/SelfCheckAlg.scala

```diff
@@ -37,12 +37,25 @@ final class SelfCheckAlg[F[_]](config: Config)(implicit
   def checkAll: F[Unit] =
     for {
       _ <- logger.info("Run self checks")
+      _ <- checkWorkspaceDirectory
       _ <- checkGitBinary
       _ <- checkScalafixBinary
       _ <- checkScalafmtBinary
       _ <- checkUrlChecker
     } yield ()
 
+  private def checkWorkspaceDirectory: F[Unit] =
+    F.whenA(!config.workspace.exists)(
```

Comment:
I've two comments:
 * `config.workspace.exists` has side-effects, we need to use `FileAlg` here
 * If the workspace exists, we should also check if it is empty.